File: Project Οντοκεντρικός Προγραμματισμός/Java Project/Hotel Project, Swing/RoomTypeB.java
public class RoomTypeB extends RoomTypeA
{
    private static double discountPerDay=0; //Μεταβλητή για έκπτωση ανά μέρα.
    //Setter
    public static void setDiscountPerDay(int aaa)
    {
        discountPerDay=aaa;
    }
    public double getPriceOfRoom()//Η μέθοδος ελέγχει ποιές μέρες είναι κλεισμένο το δωμάτιο και υπολογίζει τα εσοδά του,
    {                             //υπολογίζωντας την καθημερινή έκπτωση.
        int i=0;
        double totalEarnings; //Μεταβλητή για συνολικά έσοδα.
        double moneyEarnedFromRoom=0; //Μεταβλητή για έσοδα ανά μέρα.
        discountPerDay=0;
        for(i=1; i<31; i++)
            {
                if(roomAvailability[i]!=null)
                    {
                        moneyEarnedFromRoom += (getPricePerDay() - discountPerDay);
                        discountPerDay += 10;
                        if( (discountPerDay / getPricePerDay()) > 0.5)
                        {
                            discountPerDay = 0.5 * getPricePerDay();
                        }
                    }
            }
        return moneyEarnedFromRoom;
    }
    public boolean cancelBooking(int bookingID)//Η μέθοδος δείχνει ότι το δωμάτιο τύπου B δεν μπορεί να ακυρωθεί.
    {
        System.out.println("Can't cancel this kind of room");
        return false;
    }
}

File: Project Οντοκεντρικός Προγραμματισμός/C++ Project/Visual Studio Hotel Project/ConsoleApplication1/RoomTypeB.cpp
#include "RoomTypeB.h"

int discountPerDay = 0;//Variable for the discount per day of a room type B and initialization of it at 0.

RoomTypeB::RoomTypeB()//Default constructor for a RoomTypeB type object.
{
}

RoomTypeB::RoomTypeB(double pricePerDay,int maxPeople)//Constructor for a RoomTypeB type object.
{
	setPricePerDay(pricePerDay);
	setMaxPeople(maxPeople);
}

RoomTypeB::~RoomTypeB()//Default destructor for a RoomTypeB type object.
{
}

double RoomTypeB::getPriceOfRoom()//Function that returns the price of a room type B.
{
	int i = 0;
	double totalEarnings;
	double moneyEarnedFromRoom = 0;
	discountPerDay = 0;
	for (i = 1; i < 31; i++)
	{
		if (roomAvailability[i] != 0)
		{
			moneyEarnedFromRoom += (getPricePerDay() - discountPerDay);
			discountPerDay += 10;
			if ((discountPerDay / getPricePerDay()) > 0.5)//Calculating the discount per day for a room type B.
				discountPerDay = 0.5 * getPricePerDay();
		}
	}
	return moneyEarnedFromRoom;
}

bool RoomTypeB::cancelBooking(int bookingID)//Function that returns false when trying to cancel a booking at a room type B.
{
	std::cout << "Can't cancel room type B.";
	return false;
}